tools/rewriter/GenCallAsm.cpp

```diff
@@ -1026,8 +1037,11 @@ std::string emit_asm_wrapper(AbiSignature sig,
     | top |Top of the stack (stack grows down on AArch64). This address is
     |     |aligned to 16 bytes.
     +-----+
-    |ind  |Space for arguments passed indirectly (i.e. in memory). These arguments
-    |args |will point to this region.
+    |ind  |Space for arguments passed indirectly (i.e. in memory). Indirect arguments
+    |args |are passed as pointers into this region; for access in another compartment
+    |     |we must copy its contents and translate the pointers.
+    +-----+
+    |align|Space to align the stack back to 16 bytes after indirect arguments.
```

Comment:
Was there a good reason to not merge this with the align section between ret ptr and stack args? We can leave it as-is if it was just for implementation simplicity.

Reply:
We can't directly merge this since they're not adjacent. We could have reordered things but doing so was going to be a more complex change and I wasn't sure at the time that this was necessarily going to fix things, so I went for the less invasive change. We could file a bug to reorder these as doing so would probably save space sometimes, but I'd be surprised if there was any performance difference.